src/Raven.Server/Documents/DocumentFlags.cs

```diff
@@ -9,24 +9,17 @@ public enum DocumentFlags
 
         Artificial = 0x1,
         Versioned = 0x2,
-        Reserved1 = 0x4,
-        Reserved2 = 0x8,
+        Revision = 0x4,
+        Reserved = 0x8,
 
         FromIndex = 0x10,
-        FromVersionStorage = 0x20,
+        Reserved1 = 0x20,
         FromReplication = 0x40,
-        Reserved3 = 0x80,
+        FromSmuggler = 0x80,
 
-        HasAttachments = 0x100
-    }
-
-    [Flags]
-    public enum NonPersistentDocumentFlags
```

Comment:
I agree with @ppekrol reasoning here, and I think that if the user want to skip versioning they can disable the versioning for the duration if they really need it, or filter it during import. Better than having to do all that work all the time.